java/src/com/ibm/streamsx/rest/Element.java

```diff
@@ -37,4 +42,31 @@ void setConnection(StreamsConnection connection) {
     public final String toString() {
         return pretty.toJson(this);
     }
+    
+    private Gson refreshJson;
+    
+    /**
+     * Refresh this element.
+     * 
+     * Attributes of this object are updated from current state
+     * though the Streams REST API.
+     * 
+     * @throws IOException Error communicating with Streams.
+     */
+    public void refresh() throws IOException {
+        String response = connection().getResponseString(self);
```

Comment:
Yeah, I just added a commit that handles Metric. I should have said in the original PR that there was going to be a follow on commit.